### PR TITLE
Restore image digest pinning

### PR DESCRIPTION
**Pull request by ryantking**
Part of the image pinning logic is currently missing from master. This
PR adds that back along with the released version of Operator Manifest Tools.

**Description of the change:**

**Motivation for the change:**

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: internal/cmd/operator-sdk/generate/bundle/bundle.go [hunk 1]
```diff
@@ -343,3 +352,26 @@ func (c bundleCmd) findManagerEnvironment(col *collector.Manifests) ([]corev1.En
 		"could not find manager deployment, should have label %s=%s", managerLabel, managerLabelValue,
 	)
 }
+
+// pinImages is used to replace all image tags in the given manifests with digests
```
Comment (camilamacedo86) [inline internal/cmd/operator-sdk/generate/bundle/bundle.go:356]: WDYT about we add some test to ensure that works as should be?
WDYT about we just use the sample generated to generate a new sample such as v3/memcached-operator-digest to ensure that it works? 

Could we have here samples with some specific options as we have in Kubebuilder to do the checks?
Would that be a good idea?

Comment (ryantking) [reply internal/cmd/operator-sdk/generate/bundle/bundle.go:356]: @camilamacedo86 This PR was already approved and merged [here][https://github.com/operator-framework/operator-sdk/pull/5523], but for some reason when we merged the feature branch into master, it was missing this commit. Adding automated testing is its own separate issue due to some other issues we found.

**Review by jmrodri — APPROVED** [review]
/lgtm

**Pull request merged**